modules/apps/transfer/transfer_test.go

```diff
@@ -39,37 +39,46 @@ func NewTransferPath(chainA, chainB *ibctesting.TestChain) *ibctesting.Path {
 	return path
 }
 
-// constructs a send from chainA to chainB on the established channel/connection
-// and sends the same coin back from chainB to chainA.
+// Constructs the following sends based on the established channels/connections
+// 1 - from chainA to chainB
+// 2 - from chainB to chainC
+// 3 - from chainC to chainB
 func (suite *TransferTestSuite) TestHandleMsgTransfer() {
 	// setup between chainA and chainB
-	path := NewTransferPath(suite.chainA, suite.chainB)
-	suite.coordinator.Setup(path)
+	// NOTE:
+	// pathAtoB.EndpointA = endpoint on chainA
+	// pathAtoB.EndpointB = endpoint on chainB
+	pathAtoB := NewTransferPath(suite.chainA, suite.chainB)
+	suite.coordinator.Setup(pathAtoB)
 
-	//	originalBalance := suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), suite.chainA.SenderAccount.GetAddress(), sdk.DefaultBondDenom)
+	originalBalance := suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), suite.chainA.SenderAccount.GetAddress(), sdk.DefaultBondDenom)
 	timeoutHeight := clienttypes.NewHeight(1, 110)
 
 	amount, ok := sdk.NewIntFromString("9223372036854775808") // 2^63 (one above int64)
 	suite.Require().True(ok)
 	coinToSendToB := sdk.NewCoin(sdk.DefaultBondDenom, amount)
 
 	// send from chainA to chainB
-	msg := types.NewMsgTransfer(path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, coinToSendToB, suite.chainA.SenderAccount.GetAddress().String(), suite.chainB.SenderAccount.GetAddress().String(), timeoutHeight, 0, "")
+	msg := types.NewMsgTransfer(pathAtoB.EndpointA.ChannelConfig.PortID, pathAtoB.EndpointA.ChannelID, coinToSendToB, suite.chainA.SenderAccount.GetAddress().String(), suite.chainB.SenderAccount.GetAddress().String(), timeoutHeight, 0, "")
 	res, err := suite.chainA.SendMsgs(msg)
 	suite.Require().NoError(err) // message committed
 
 	packet, err := ibctesting.ParsePacketFromEvents(res.GetEvents())
 	suite.Require().NoError(err)
 
 	// relay send
-	err = path.RelayPacket(packet)
+	err = pathAtoB.RelayPacket(packet)
 	suite.Require().NoError(err) // relay committed
 
+	// check that module account escrow address has locked the tokens
+	escrowAddress := types.GetEscrowAddress(packet.GetSourcePort(), packet.GetSourceChannel())
+	balance := suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), escrowAddress, sdk.DefaultBondDenom)
+	suite.Require().Equal(coinToSendToB, balance)
```

Comment:
@bugarela I added the check for the escrow account on chain A. I hope this is what you were trying to add.

Reply:
Thanks! Makes sense to me